pkg/reconciler/v1alpha1/pipelinerun/resources/conditionresolution.go

```diff
@@ -95,11 +99,20 @@ func (rcc *ResolvedConditionCheck) ConditionToTaskSpec() *v1alpha1.TaskSpec {
 		t.Inputs = &v1alpha1.Inputs{
 			Params: rcc.Condition.Spec.Params,
 		}
+		convertParamTemplates(&t.Steps[0], rcc.Condition.Spec.Params)
```

Comment:
Why is this needed? The comment on `convertParamTemplates` explains what it does but I'm still a bit lost on the reason we do it.

Reply:
oh wait is this because the condition spec places the params at a higher level than the `ApplyContainerReplacements` func is expecting and we're trying to reuse that logic from taskruns?  might be worth adding a comment to that effect here.

Reply:
added!